### PR TITLE
Support V2 and V3 calling simultaneously

### PR DESCRIPTION
**Pull request by typfel**
Implements the CallFlow methods on the V3 voice channel and redirect the V2 related methods to the V2 voice channel.

Needs release from:
- [x]  https://github.com/wireapp/wire-ios-data-model/pull/89

### CODE REVIEW

File: Source/Calling/VoiceChanneRouter+CallFlow.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    }
+    
+    @objc(toggleVideoActive:error:)
+    public func toggleVideo(active: Bool) throws {
+        try setVideoSendActive(active)
+    }
+    
+    public func join(video: Bool) {
+        if video {
+            try? joinVideoCall() // FIXME
```
Comment (daehn) [inline Source/Calling/VoiceChanneRouter+CallFlow.swift]: `FIXME` left

File: Source/Calling/ZMCallKitDelegate.m [hunk 2]
```diff
@@ -668,11 +668,12 @@ - (void)conversationVoiceChannelStateDidChange:(ZMConversation *)conversation pr
     case ZMVoiceChannelStateSelfConnectedToActiveChannel:
             [self.provider reportOutgoingCallWithUUID:conversation.remoteIdentifier
                                       connectedAtDate:[NSDate date]];
-            conversation.voiceChannel.callStartDate = [NSDate date];
+            // FIXME
```
Comment (daehn) [inline Source/Calling/ZMCallKitDelegate.m]: `FIXME` left

File: Source/Calling/ZMCallKitDelegate.m [hunk 3]
```diff
@@ -696,7 +697,8 @@ - (void)conversationVoiceChannelStateDidChange:(ZMConversation *)conversation pr
                                       endedAtDate:nil
                                            reason:CallEndedReasonFromZMVoiceChannelState(conversation.voiceChannel.state, prevState)];
             }
-            conversation.voiceChannel.callStartDate = nil;
+            // FIXME
```
Comment (daehn) [inline Source/Calling/ZMCallKitDelegate.m]: I guess I will add this rule to `Danger` so it does not only find todos

File: Tests/Source/Calling/ZMFlowSyncTests.m [hunk 4]
```diff
@@ -21,9 +21,10 @@
 @import ZMUtilities;
 @import ZMCDataModel;
 @import zmessaging;
+@import avs;
 
 #import "MessagingTest.h"
-#import "AVSFlowManager.h"
+//#import "AVSFlowManager.h"
```
Comment (daehn) [inline Tests/Source/Calling/ZMFlowSyncTests.m]: Can be removed?

File: Source/UserSession/ZMUserSession+Background.m [hunk 5]
```diff
@@ -332,12 +330,12 @@ - (void)handleConnectionRequestCategoryNotification:(ZMStoredLocalNotification *
 - (void)handleCallCategoryNotification:(ZMStoredLocalNotification *)note
 {
     if (note.actionIdentifier == nil || [note.actionIdentifier isEqualToString:ZMCallAcceptAction]) {
-        BOOL callIsStillOngoing = (note.conversation.callParticipants.count > 0);
+        BOOL callIsStillOngoing = (note.conversation.callParticipants.count > 0) || note.conversation.voiceChannel.state == ZMVoiceChannelStateIncomingCall; // FIXME check if this logic works
```
Comment (daehn) [inline Source/UserSession/ZMUserSession+Background.m]: `FIXME`

File: Tests/Source/Integration/CallingTests.m [hunk 6]
```diff
@@ -2937,7 +2937,28 @@ - (void)testThatItLogsEverything
 @end
 
 
+@implementation CallingTests (CallingV3)
 
+- (void)testThatItCallingV3SendingMessage;
```
Comment (daehn) [inline Tests/Source/Integration/CallingTests.m]: Maybe we should test the behaviour in error conditions as well

**Comment by typfel** [comment]
Not every `FIXME` will be solved in this PR.

**Pull request merged**